src/components/pages/CheckoutPage/CheckoutPage.jsx

```diff
@@ -17,23 +17,321 @@
+                    }
+                })
+                .then(res => res.json())
+                .then(res => {
+                    this.setState({
+                        orderData: res
+                    }, () => { console.log(this.state.orderData) });
+                })
+                .catch(error => {
+                    console.log(error)
+                });
+        });
+    }
+    componentDidMount() {
+        this.fetchOrderData();
+    }
+    newAddress() {
+        this.props.history.push('/newaddressform', { returnPage: '/checkout' });
+    }
+    editAddress(addressLink) {
+        this.props.history.push('/editaddress', { returnPage: '/checkout', address: addressLink });
+    }
+    deleteAddress(addressLink) {
+        login().then(() => {
+            fetch(addressLink, {
+                method: 'delete',
+                headers: {
+                    'Content-Type': 'application/json',
+                    'Authorization': localStorage.getItem(Config.cortexApi.scope + '_oAuthToken')
+                }
+            }).then(() => {
+                this.fetchOrderData();
+            }).catch(error => {
+                console.log(error);
+            });
+        });
+    }
+    handleChange(link) {
+        login().then(() => {
+            fetch(link, {
+                method: 'post',
+                headers: {
+                    'Content-Type': 'application/json',
+                    'Authorization': localStorage.getItem(Config.cortexApi.scope + '_oAuthToken')
+                }
+            }).then(() => {
+                this.fetchOrderData();
+            }).catch(error => {
+                console.log(error);
+            })
+        })
+    }
+    newPayment() {
+        this.props.history.push('/newpaymentform', { returnPage: '/checkout' });
+    }
+    renderShippingAddress() {
+        if (this.state.orderData['_order'][0]['_deliveries'] && this.state.orderData['_order'][0]['_deliveries'][0]['_element'][0]['_destinationinfo']) {
+            var shippingAddresses = this.state.orderData['_order'][0]['_deliveries'][0]['_element'][0]['_destinationinfo'][0]['_selector'][0]['_chosen'];
+            if (this.state.orderData['_order'][0]['_deliveries'][0]['_element'][0]['_destinationinfo'][0]['_selector'][0]['_choice']) {
+                const choices = this.state.orderData['_order'][0]['_deliveries'][0]['_element'][0]['_destinationinfo'][0]['_selector'][0]['_choice'];
+                shippingAddresses = shippingAddresses.concat(choices);
+            }
+            return (
+                shippingAddresses.map((shippingAddress, index) => {
+                    const name = shippingAddress['_description'][0]['name'];
+                    const address = shippingAddress['_description'][0]['address'];
+                    return (
+                        <div key={"shippingOption_" + Math.random().toString(36).substr(2, 9)}>
```

Comment:
Is there any other key you can use? Any identifier for the address?

Reply:
there is no identifier for the address :( I'm also purposely using random here so that when I select a different address, when it is re-rendered it appears in the top position and as 'selected'. when I used index as the key, when selecting a new address, the 'selected' would be in the wrong position.